### PR TITLE
Always use vector for arrays

### PR DESCRIPTION
**Pull request by vadz**
The main change here is getting rid of all the dynamic array legacy code and always using either `wxVector<>` (still the default, although I wonder if it's still worth it) or `std::vector<>`. AFAICS this should be quite compatible, so I don't see any reason to not make this change.

And a nice side effect is that now by removing the allocation size limit in `wxVector<>` we can speed up not only the code using it, but also all the existing code using `wxArrayString` too.

### CODE REVIEW

File: include/wx/dynarray.h [hunk 1]
```diff
@@ -24,8 +24,12 @@
     template <typename T>
     inline void wxVectorShrinkToFit(std::vector<T>& v)
     {
+    #if __cplusplus >= 201103L
```
Comment (MaartenBent) [inline include/wx/dynarray.h]: Maybe add a check for MSVC as well, it is still at `199711L`.
E.g. as in `sting.h`:
```c++
#if (__cplusplus >= 201103L || wxCHECK_VISUALC_VERSION(10))
```

Comment (vadz) [reply include/wx/dynarray.h]: Yes, thanks for noticing this.

**Comment by vadz** [comment]
Unfortunately I have to abandon this idea because of several compatibility problems uncovered by the "Always used vector-based implementation for dynamic arrays" commit. This is really disappointing, but fixing them probably requires rewriting all the legacy macros as templates, which would be well worth doing but requires more time than I can possibly spend on it now.

**Comment by eranif** [comment]
How about just fixing the `wxArrayString::Add` memory grow policy?

**Comment by eranif** [comment]
Sorry :/ I missed this commit: https://github.com/wxWidgets/wxWidgets/commit/ac052d60234680a4806609d6529e04e7622f8f31

**Pull request closed without merging**